src/RefractAST.cc

```diff
@@ -194,47 +194,75 @@ namespace drafter
 
     static refract::IElement* MsonElementToRefract(const mson::Element& mse);
 
-    // FIXME: check against original behavioration
-    template <typename T, typename V = typename T::ValueType>
+    RefractElements MsonElementsToRefract(const mson::Elements& elements) {
+        RefractElements result;
+        std::transform(elements.begin(), elements.end(), std::back_inserter(result), MsonElementToRefract);
+        return result;
+    }
+
+    struct TypeSectionData {
+        RefractElements defaults;
+        RefractElements samples;
+        std::vector<std::string> descriptions;
+    };
+
+    template <typename T>
     struct ExtractTypeSection
     { // This will handle primitive elements
-        const mson::TypeSection& ts;
+        std::vector<typename T::ValueType> values;
+        TypeSectionData& data;
 
-        ExtractTypeSection(const mson::TypeSection& v) : ts(v)
-        {
-        }
+        template <typename U, bool dummy = true>
```

Comment:
This `dummy` param is required as workaround for inner template specialization
will add explaining comment